doc/reference/kernel/other/float.rst

```diff
@@ -100,6 +100,31 @@ If an ARM thread does not require use of the floating point registers any
 more, it can call :cpp:func:`k_float_disable()`. This instructs the kernel
 not to save or restore its FP context during thread context switching.
 
+On the ARCv2 architecture the kernel treats each thread as a non-user,
```

Comment:
Now that we have 3 architectures, here, perhaps it would be nice for the reader to see separate paragraphs, for ARC, ARM and x86 Shared FP services. So they can easily focus on the ARCH they want. @dbkinder what do you think?

Reply:
I think adding some subheadings would help here, before each section that talks about a specific architecture:

- before line 74 add:
  ```
  ARM Cortex-M architecture (with the Floating Point Extension)
  -------------------------------------------------------------
  ```
- before line 103 add:
  ```
  ARCv2 architecture
  ------------------
  ```
- and before line 128 add:
  ```
  x86 architecture
  ----------------
  ```